api/v1beta1/awsmachinetemplate_webhook.go

```diff
@@ -113,6 +115,18 @@ func (r *AWSMachineTemplate) ValidateCreate() error {
 	allErrs = append(allErrs, r.validateRootVolume()...)
 	allErrs = append(allErrs, r.validateNonRootVolumes()...)
 
+	// Feature gate is not enabled but ignition is enabled then send a forbidden error.
+	if !feature.Gates.Enabled(feature.BootstrapFormatIgnition) && spec.Ignition != nil {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "ignition"),
+			"can be set only if the BootstrapFormatIgnition feature gate is enabled"))
+	}
+
+	cloudInitConfigured := spec.CloudInit.SecureSecretsBackend != "" || spec.CloudInit.InsecureSkipSecretsManager
```

Comment:
Same comment with cloudInitConfigured(): this check is not providing additional value as both fields can be unset but still cloudInit is being used.

Reply:
Thanks for pointing out. Let's discuss it in https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2271#discussion_r826311478, I'll mark this resolved.